### PR TITLE
Set module to Java.base before java.base is created.

### LINKED ISSUE #2839: Are lambdas for java.base defined in the unnamed module?
I built openJ9 myself on my local machine and added the following log line in createramclass.cpp~2810 after
module = javaVM->unamedModuleForSystemLoader;
printf("creatramclass.internalCreateRAMClassFromROMClass.unamed %s\n", J9UTF8_DATA(className));

Now in the log i see the following line:
creatramclass.internalCreateRAMClassFromROMClass.unamed jdk/internal/loader/BuiltinClassLoader$$Lambda$1/000000009B3D8510

So I got a question, are the lambdas for java.base classes like (BuiltinClassLoader) defined in unnamed module, or I am reading something wrong.

### PR DESCRIPTION
**Pull request by hangshao0**
Fixes #2839

Signed-off-by: hangshao <hangshao@ca.ibm.com>

### CODE REVIEW

File: runtime/vm/createramclass.cpp [hunk 1]
```diff
@@ -2803,14 +2803,7 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
...
 			} else {
-				if (J9ROMCLASS_IS_PRIMITIVE_TYPE(romClass)
-					|| (LOAD_LOCATION_PATCH_PATH == locationType)
-					|| (LOAD_LOCATION_MODULE == locationType)
-				) {
-					module = javaVM->javaBaseModule;
-				} else {
-					module = javaVM->unamedModuleForSystemLoader;
-				}
+				module = javaVM->javaBaseModule;
```
Comment (DanHeidinga) [inline runtime/vm/createramclass.cpp]: Can you add a comment explaining why we chose this behaviour?  Once there's a comment here I'm good with the change.

Comment (hangshao0) [reply runtime/vm/createramclass.cpp]: Comment added

File: runtime/vm/createramclass.cpp [hunk 2]
```diff
@@ -2803,14 +2803,10 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
...
-					|| (LOAD_LOCATION_PATCH_PATH == locationType)
-					|| (LOAD_LOCATION_MODULE == locationType)
-				) {
-					module = javaVM->javaBaseModule;
-				} else {
-					module = javaVM->unamedModuleForSystemLoader;
-				}
+				/* Do not assign module based on locationType, as we never set locationType for anonymous classes which led us to assign them to
+				 * unamedModuleForSystemLoader incorrectly. Change the code to directly assign classes to Java.base here.
+				 */
```
Comment (DanHeidinga) [inline runtime/vm/createramclass.cpp]: This isn't quite the comment I was expecting as the situation is more broad than anonymous classes, which are correctly assigned the `hostClass->module` after java.base is created.

Maybe something more along the lines of:
```
Ignore locationType and assign all classes created before the java.base module is created to java.base.
This matches the RI implementation.  Validated on JDK9 through JDK11.
```

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk10

**Comment by hangshao0** [comment]
@DanHeidinga , seems that sanity plinux jdk10 build is not triggered.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk10

**Comment by DanHeidinga** [comment]
@AdamBrousseau The PR builds don't appear to work on this PR.  Any insight?

**Comment by AdamBrousseau** [comment]
Jenkins copyright check

**Comment by AdamBrousseau** [comment]
Jenkins copyright check

**Comment by AdamBrousseau** [comment]
Jenkins line endings check

**Comment by AdamBrousseau** [comment]
Jenkins test sanity plinux jdk10

**Pull request merged**